### PR TITLE
Added E2E for vmock auto-completion.

### PR DESCRIPTION
**Pull request by Den1552**
## Summary

Added E2E for vmock auto-completion, where the release get switched when opening vscode. 

## Added
- `vcast_coded_test_different_envs_hover.test`, including the tests.
- Spec group `coded_mock_different_env`.

## Changed
- Refactored `wdio.conf.ts` for better maintainability.

### CODE REVIEW

File: tests/internal/e2e/test/specs/vcast_coded_test_different_envs_hover.test.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  getViewContent,
+  findSubprogram,
+  getTestHandle,
+  updateTestID,
+  normalizeContentAssistString,
+  getGeneratedTooltipTextAt,
+} from "../test_utils/vcast_utils";
+
+// Define the normalized version of the expected content
+export const normalizedExpectedFunctionOutput = `
```
[the remark this answers is not in the document]
Comment (Den1552) [reply tests/internal/e2e/test/specs/vcast_coded_test_different_envs_hover.test.ts]: Deleted in b7bee41b81ade080cc61d605bbe84ea5603a44b4 .

**Review by aytey — APPROVED** [review]

File: tests/internal/e2e/test/wdio.conf.ts [hunk 2]
```diff
@@ -443,11 +444,28 @@ export const config: Options.Testrunner = {
         "vcastTutorial"
       );
 
-      const vcastRoot = await getVcastRoot();
-      const newVersion = "2024sp4";
+      let vcastRoot = await getVcastRoot();
+      const coded_mock_different_env_version = "2024sp1";
+      const newestRelease = "2024sp4";
```
Comment (Zbigor) [inline tests/internal/e2e/test/wdio.conf.ts:449]: could we pick this up from the environment variables somehow (from those github actions env vars maybe)? Or could we define this "centrally" somewhere in the `spec_config.ts`?

**Review by Zbigor — APPROVED** [review]
Just a minor comment, looks good otherwise

File: tests/internal/e2e/test/wdio.conf.ts (hunk 2 above)
Comment (Den1552) [reply tests/internal/e2e/test/wdio.conf.ts]: Exported from `specs_config.ts` in ce889f2f2a4161f8ff360a079bfe45911588f16e.

**Pull request merged**